### ISSUE TITLE
xWebConfigPropertyCollection: Timing issue in integration tests

### ISSUE THREAD

**Issue by johlju** [PowerShell/xWebAdministration#385]
There seems to be a timing issue with one of the integration tests in xWebConfigPropertyCollection. The integration tests passed on the PR, but failed when dev-branch ran tests.

```
IOException: The process cannot access the file 'C:\windows\system32\inetsrv\config\applicationHost.config' because it is being used by another process.
    at <ScriptBlock>, C:\projects\xwebadministration\Tests\Integration\MSFT_xWebConfigPropertyCollection.Integration.Tests.ps1: line 203
    at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\Pester\4.4.0\Pester.psm1: line 839
    at Invoke-Pester<End>, C:\Program Files\WindowsPowerShell\Modules\Pester\4.4.0\Pester.psm1: line 854
    at Invoke-AppveyorTestScriptTask, C:\projects\xwebadministration\DscResource.Tests\AppVeyor.psm1: line 796
Executing script C:\projects\xwebadministration\Tests\Integration\MSFT_xWebsite.Integration.Tests.ps1
There is no operation running currently. Stop will return without any action.
```
https://ci.appveyor.com/project/PowerShell/xwebadministration/build/1.17.1123.0?fullLog=true#L2292

It seems it fails on this row.
https://github.com/PowerShell/xWebAdministration/blob/50177b94684223cda326989a57fd0d321d378066/Tests/Integration/MSFT_xWebConfigPropertyCollection.Integration.Tests.ps1#L203

Can we do something to prevent this from happening?

**Comment by johlju** [comment]
Note: If the tests fail when we release, then this will become a blocking release issue.

**Comment by regedit32** [comment]
I haven't been able to replicate the issue in my lab, but I've seen it happen occasionally on the TFS build my team uses.  A few weeks back I might have resolved it by rearranging the finally block like this below, but I can't recall if it has popped up again.

```
    #region FOOTER
    Restore-TestEnvironment -TestEnvironment $TestEnvironment

    Restore-WebConfiguration -Name $tempName
    Remove-WebConfigurationBackup -Name $tempName
    #endregion
```

By running Restore-TestEnvironment first, it just gives the system an extra second to unlock the file handle. I'd imagine an arbitrary 5 second start-sleep would work too. 

Do we need a more robust solution than that?

**Comment by regedit32** [comment]
This is also the same as #377

**Comment by johlju** [comment]
I think that will do, since this is the integration tests, and we are restoring the test environment, a `Start-Sleep` a few seconds is good with me. 

Closed issue #377 as a duplicate, since we are discussing here.
